### PR TITLE
add events module

### PR DESCRIPTION
**Pull request by sainak**
## Proposed Changes

- Brief of changes made.

### Associated Issue

- Link to issue here, explain how the proposed solution will solve the reported issue/ feature request.

### Architecture changes

- Remove this section if not used

## Merge Checklist

- [ ] Tests added/fixed
- [ ] Update docs in `/docs`
- [ ] Linting Complete
- [ ] Any other necessary step

_*Only PR's with test cases included and passing lint and test pipelines will be reviewed*_

@coronasafe/care-backend-maintainers @coronasafe/care-backend-admins

### CODE REVIEW

File: care/facility/models/events.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        if self.description is not None and not self.description.strip():
+            self.description = None
+        return super().save(*args, **kwargs)
+
+    def __str__(self) -> str:
+        return f"{self.model} - {self.name}"
+
+
+class PatientConsultationEvent(models.Model):
+    # external_id = use ulid
```
Comment (vigneshhari) [inline care/facility/models/events.py:48]: We need some sort of non integer id for each event, i recommend making that the primary key

File: care/facility/events/handler.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    caused_by: int,
+    created_date: datetime = None,
+    old: Model = None,
+):
+    if created_date is None:
+        created_date = now()
+
+    if isinstance(objects, (QuerySet, list, tuple)):
+        if old is not None:
+            raise ValueError("diff is not available when objects is a list or queryset")
```
Comment (vigneshhari) [inline care/facility/events/handler.py:114]: Should we just log or raise a sentry issue for this for now ?

File: care/facility/events/handler.py [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+        for obj in objects:
+            object_model = obj.__class__.__name__
+            create_consultation_event_task.apply_async(
```
Comment (vigneshhari) [inline care/facility/events/handler.py:117]: Events should always be generated live inside a transaction, No background worker should be involved.

Comment (vigneshhari) [reply care/facility/events/handler.py:117]: We want to be 100% certain that a change has an event associated with it. let's build it that way from the beginning.

File: care/facility/events/handler.py [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+    consultation_id: int,
+    object_model: str,
+    object_id: int,
+    caused_by: int,
+    created_date: datetime,
+    diff: str = None,
+):
+    Model = apps.get_model("facility", object_model)
+
+    print(f"Creating Event for {object_model} {object_id}")
```
Comment (vigneshhari) [inline care/facility/events/handler.py:46]: Remove all instances of print

Comment (vigneshhari) [reply care/facility/events/handler.py:46]: Log if you really need to. ( chances are you don't  )

**Review by vigneshhari — CHANGES_REQUESTED** [review]

File: care/facility/events/handler.py (hunk 2 above)
Comment (sainak) [reply care/facility/events/handler.py:108]: this is a parameter validation for the dev

**Review by vigneshhari — APPROVED** [review]

**Pull request merged**